StylePropertyMap implemented the append method

njlawton opened the pull request.

Commit:
Added the _allowsListValuesdictionary to the PropertyDictionary class
This holds CSS Style values that can accept lists of StyleValues as input as well as the string separator they use to separate each StyleValue e.g. , or a space. the CSS property animation-iteration-count was added to PropertyDictionary as it supports lists of StyleValues.

Commit:
Implemented helper methods relating to the _allowsListValues dictionary
isListValuedProperty and getListValueSeparator have been implemented in the PropertyDictionary class as well as test to validate its function

Commit:
Fixed incorrect keyword in PropertyDictionary class

Commit:
Implemented the append method in the StylePropertyMap class
The method has been implemented and tests have been implemented to validate the methods behaviour

Commit:
Moved the throwInvalidInputError method to the PropertyDictionary class
This method was moved as it made more sense to be placed here the method was also changed to public

Commit:
Improved error messages and there test cases for the append method
Test cases now test that the correct message is printed this is improtant as all errors throwan are TypeErrors but the errors themselves are for different reasons

Commit:
Fixed append method for the case were you are appending to an unset property

Commit:
removed a hard coded value in the append method
This was an accedental addition and was hence removed and replaced with the correct value

Commit:
Changed the wording of append method test and error messages

Commit:
Fixed test case with wrong error message

wilddamon commented on src/property-dictionary.js:
```diff
@@ -18,24 +18,43 @@
     };
+
+    this._allowsListValues = {
+      'animationIterationCount': ', '
```
What does the comma mean here? Should it just be true like _allowsPercentage?

[edit] Oh I see. To prevent conclusion, perhaps this should be called _listSeparator or something similar.

wilddamon commented on src/property-dictionary.js:
```diff
@@ -18,24 +18,43 @@
+    if (this.isListValuedProperty(property)) {
+      return this._allowsListValues[property];
+    }
+    throw new TypeError(property + ' does not support lists of styleValues');
```
s/styleValues/StyleValues (capitalization)

wilddamon commented on src/style-property-map.js:
```diff
@@ -36,22 +36,58 @@
+      }
+      if (cssAppendString == '') {
+        cssAppendString += values[i].cssString;
+        continue;
```
Perhaps this should be if/else to prevent confusion. The `continue` statement threw me a little.

wilddamon commented on src/style-property-map.js:
```diff
@@ -36,22 +36,58 @@
   StylePropertyMap.prototype.delete = function(property) {
-    throw new TypeError('Function not implemented yet');
+    var cssPropertyDictionary = propertyDictionary();
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
```
You can simplify these two lines by writing

``` javascript
if (!propertyDictionary().isSupportedProperty(property)) {
```

It's basically the same length...

wilddamon commented on src/style-property-map.js:
```diff
@@ -36,22 +36,58 @@
+
+  StylePropertyMap.prototype.has = function(property) {
+    var cssPropertyDictionary = propertyDictionary();
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
```
Same here.

wilddamon commented on src/style-property-map.js:
```diff
@@ -36,22 +36,58 @@
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
+      throw new TypeError(property + ' is not a supported CSS property');
+    }
+    return !(this._styleObject[property] == '');
```
You can just write

``` javascript
return !!this._styleObject[property]
```

"", null, undefined and 0 all evaluate to false, most other things evaluate to true, and using `!!` coerces the value to be a boolean.

Commit:
Made improvements to the readablity of code for the has and delete methods
The changes made to thease methods in the StylePropertyMap reduses un necassary lines and improves understandability of the code

Commit:
Merge branch 'master' into Style-map-append

wilddamon commented on src/property-dictionary.js:
```diff
@@ -19,25 +19,42 @@
      'border-top-width': [LengthValue],
-     'opacity': [NumberValue]
+     'opacity': [NumberValue],
+     'animationIterationCount': [NumberValue]
```
to be consistent, use `animation-iteration-count`

Commit:
Changed animationIterationCount to animation-iteration-count for consistancy

wilddamon commented:
LGTM
